### ISSUE TITLE
Plugin Disabled

### ISSUE THREAD

**Issue by BenBranny** [ryandw11/CustomStructures#14]
I installed the plugin and it worked fine (with commands) I hadn't created any structures yet. However, after saving the schematic and sorting out the loot tables it no longer worked. I decided to retry by completely reinstalling the plugin again but this time it still didn't work, even after a fresh reinstall. The only issue I could see this being is having a world edit saved schematic as that's the only change I've made to my server. Could you please help me with how to fix it.

**Comment by ryandw11** [comment]
Can I see you config? Are there any errors in the console on startup? What minecraft version are you on?

**Comment by BenBranny** [comment]
This is what i currently have setup



Version: Spigot 1.14.4 (git-Spigot-9de398a-9c887d4)



Console:
23.07 17:39:07 [Server] Server thread/INFO BenBranny issued server command: /customstructure
23.07 17:39:07 [Server] Server thread/ERROR null
23.07 17:39:07 [Server] INFO org.bukkit.command.CommandException: Cannot execute command 'customstructure' in plugin CustomStructures v1.4.1 - plugin is disabled.
23.07 17:39:07 [Server] INFO at org.bukkit.command.PluginCommand.execute(PluginCommand.java:37) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at org.bukkit.craftbukkit.v1_14_R1.CraftServer.dispatchCommand(CraftServer.java:710) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.PlayerConnection.handleCommand(PlayerConnection.java:1641) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:1481) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.PacketPlayInChat.a(PacketPlayInChat.java:47) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.PacketPlayInChat.a(PacketPlayInChat.java:1) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.aX(MinecraftServer.java:910) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:903) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.sleepForTick(MinecraftServer.java:887) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:820) [spigot_1.14.4.jar:git-Spigot-9de398a-9c887d4]
23.07 17:39:07 [Server] INFO at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]



Config.yml:
configversion: 3
#========================================
#             Custom Structures    
#========================================
#Wiki: https://github.com/ryandw11/CustomStructures/wiki

#Allow Plugins to delete structures at will.
allowDeletion: true

Schematics:
    basicDungeon:
        Schematic: 'BasicDungeon.schem'
        Biome: 'all'
        Chance:
            Number: 1
            OutOf: 1000
        AllWorlds: false
        AllowedWorlds:
            - world
        SpawnY: top
        PlaceAir: true
        spawnInLiquid: false
        randomRotation: true
        whitelistSpawnBlocks:
            - Grass_block
            - Dirt
            - Stone
            - air
        LootTables:
            lootTableBasicDungeon: 11



lootTableBasicDungeon.yml:
#LootTable Example
#You can create as many files as you want, each file represents a loot table

#Must be ENTITY for loot on entity drops or CHEST for loot on treasure chest
Type: 'CHEST'
#Determines how many Items will be chosen from the Items list bellow
Rolls:  5
Items:
  item0:
    #Material Type https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html
    Type: 'IRON_SWORD'
    #Number of items per stack
    Amount: 1
    #Determines how often this entry will be chosen out of all the entries in the list.
    Weight: 10
  item1:
    Type: 'IRON_HELMET'
    Amount: 1
    Weight: 5
  item2:
    Type: 'ROTTEN_FLESH'
    Amount: 5
    Weight: 20
  item3:
    Name: '&cDungeon Slayer'
    Type: 'IRON_SWORD'
    Amount: 1
    Weight: 10
    Enchantments:
      FIRE_ASPECT: 1
      DAMAGE_ALL: 1
  item3:
    Name: '&cBane Of Dungeon'
    Type: 'DIAMOND_SWORD'
    Amount: 1
    Weight: 1
    Enchantments:
      FIRE_ASPECT: 1
      DAMAGE_ALL: 4
      LOOT_BONUS_MOBS: 2
    item4:
      Type: 'LEATHER_HELMET'
      Amount: 1
      Weight: 20
    item5:
      Type: 'LEATHER_BOOTS'
      Amount: 1
      Weight: 15
      Enchantments:
        PROTECTION_ENVIRONMENTAL: 1
    item6:
      Type: 'LEATHER_LEGGINGS'
      Amount: 1
      Weight: 15
      Enchantments:
        THORNS: 2
    item7:
      Type: 'LEATHER_CHESTPLATE'
      Amount: 1
      Weight: 20
    item8:
      Type: 'STRING'
      Amount: 12
      Weight: 20
    item9:
      Type: 'LEATHER_BOOTS'
      Amount: 7
      Weight: 20
    item10:
      Type: 'BONE'
      Amount: 12
      Weight: 20
    item11:
      Type: 'STRING'
      Amount: 17
      Weight: 18
    item12:
      Type: 'BONE'
      Amount: 23
      Weight: 18



![Screenshot (10)](https://user-images.githubusercontent.com/10486128/61730818-19120880-ad72-11e9-93aa-48dab90f7238.png)
![Screenshot (11)](https://user-images.githubusercontent.com/10486128/61730819-19120880-ad72-11e9-9aad-97cadeb62771.png)



I hope this is enough to help.

**Comment by ryandw11** [comment]
Your tabbing is messed up in  the loottable. Can you put the yaml config in a pastebin or something.

**Comment by BenBranny** [comment]
https://pastebin.com/T6kV6Cpx

Here you go!

**Comment by BenBranny** [comment]
UPDATE: I fixed the repeated item3 name and tabbing (i think) but it still doesn't work :(

**Comment by ryandw11** [comment]
You were missing a qutote mark on one of the items.

```yml
#LootTable Example
#You can create as many files as you want, each file represents a loot table
 
#Must be ENTITY for loot on entity drops or CHEST for loot on treasure chest
Type: 'CHEST'
#Determines how many Items will be chosen from the Items list bellow
Rolls: 5
Items:
  item0:
    #Material Type https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html
    Type: 'IRON_SWORD'
    #Number of items per stack
    Amount: 1
    #Determines how often this entry will be chosen out of all the entries in the list.
    Weight: 10
  item1:
    Type: 'IRON_HELMET'
    Amount: 1
    Weight: 5
  item2:
    Type: 'ROTTEN_FLESH'
    Amount: 5
    Weight: 20
  item:
    Name: '&cDungeon Slayer'
    Type: 'IRON_SWORD'
    Amount: 1
    Weight: 10
    Enchantments:
      FIRE_ASPECT: 1
      DAMAGE_ALL: 1
  item3:
    Name: '&cBane Of Dungeon'
    Type: 'DIAMOND_SWORD'
    Amount: 1
    Weight: 1
    Enchantments:
      FIRE_ASPECT: 1
      DAMAGE_ALL: 4
      LOOT_BONUS_MOBS: 2
  item4:
    Type: 'LEATHER_HELMET'
    Amount: 1
    Weight: 20
  item5:
    Type: 'LEATHER_BOOTS'
    Amount: 1
    Weight: 15
    Enchantments:
      PROTECTION_ENVIRONMENTAL: 1
  item6:
    Type: 'LEATHER_LEGGINGS'
    Amount: 1
    Weight: 15
    Enchantments:
      THORNS: 2
  item7:
    Type: 'LEATHER_CHESTPLATE'
    Amount: 1
    Weight: 20
  item8:
    Type: 'STRING'
    Amount: 12
    Weight: 20
  item9:
    Type: 'LEATHER_BOOTS'
    Amount: 7
    Weight: 20
  item10:
    Type: 'BONE'
    Amount: 12
    Weight: 20
  item11:
    Type: 'STRING'
    Amount: 17
    Weight: 18
  item12:
    Type: 'BONE'
    Amount: 23
    Weight: 18
```

**Comment by BenBranny** [comment]
I've tried the above but it still says disabled

**Comment by ryandw11** [comment]
On server start up there is a message when the plugin is disabled. That message tells you what happened and why the plugin is disabled. Can you show me that message?

**Comment by BenBranny** [comment]
sorry for the late reply here's the console error:

https://pastebin.com/MzVH4N7y

**Comment by ryandw11** [comment]
Can you send me your entire log. The plugin displays the message right as the server is starting up.

**Comment by BenBranny** [comment]
I looked at the error and it said that the schematic wasn't there so I'm not sure how it got deleted but I have re-copied and pasted it into the schematics folder now the plugin doesn't disable on start. However, I'm trying to test a structure by doing /cs test BasicDungeon.schem and it says in chat "script does not exist: /home/minecraft/multicraft/servers/server126279/plugins/WorldEdit/craftscripts/test.js" and I'm not really sure what this means or how to fix it.

**Comment by BenBranny** [comment]
heres the new console log btw
https://pastebin.com/sVVHYc5c

**Comment by ryandw11** [comment]
/cs is a worldedit command. You want to use /ctructure instead

**Comment by BenBranny** [comment]
We've made progress! The structure now spawns in but there is no loot inside the chests. Here are the links to the config.yml and lootTableBasicDungeon.yml. I have also tried in the config.yml, removing the .yml when it says lootTables: lootTableBasicDungeon.yml. I hope you understand what I meant.

https://pastebin.com/39aUMV1B
https://pastebin.com/8RMmrBWq

**Comment by ryandw11** [comment]
You have `randomRotation: true`. Set it to false. There is currently a bug where loottables do not work with random rotations.

**Comment by BenBranny** [comment]
I have done that but it still doesn't spawn anything in the chests

**Comment by ryandw11** [comment]
Yea you need to now remove the .yml from lootTableBasicDungeon.yml

**Comment by BenBranny** [comment]
I've done that but it still doesn't work.

**Comment by BenBranny** [comment]
Also, could u tell me by 1/1000 rarity is that like per block or chunk or what

**Comment by ryandw11** [comment]
There is a 1/1000 chance for the structure to spawn every time a chunk loads for the first time. So for it means that for approximately 1000 chunks 1 will spawn.

**Comment by BenBranny** [comment]
And could u tell me what the issue is about why the loot won’t spawn now

**Comment by ryandw11** [comment]
Do you have randomrotation enabled?

**Comment by BenBranny** [comment]
No I’ve set that to false but the issue still occurs

**Comment by ryandw11** [comment]
Any new errors in the log?

**Comment by BenBranny** [comment]
There doesn't seem to be any new errors but here is my current log: https://pastebin.com/kSRKZauh

Config.yml:
https://pastebin.com/MCfvLZhx

lootTableBasicDungeon.yml:
https://pastebin.com/C8TKqS8b

**Comment by diomonder** [comment]
> There doesn't seem to be any new errors but here is my current log: https://pastebin.com/kSRKZauh
> 
> Config.yml:
> https://pastebin.com/MCfvLZhx
> 
> lootTableBasicDungeon.yml:
> https://pastebin.com/C8TKqS8b

Im having the same problem with loot not spawning with a near identical setup.

**Comment by ryandw11** [comment]
Interesting. Your logs don't have the confirmation message saying that the loot tables were ok. I will look further into it.

**Comment by ryandw11** [comment]
I discovered the issue I believe. This will be patched in 1.4.2.1. If this bug continues after that please open a new issue with the new errors. Thanks for reporting!
